packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart

```diff
@@ -45,34 +93,63 @@ class GoogleSignInPlugin {
     }
   }
 
+  Map<String, dynamic> _currentUserToPluginMap(GoogleUser currentUser) {
+    assert(currentUser != null);
+    final Auth2BasicProfile profile = currentUser.getBasicProfile();
+    return <String, dynamic>{
+      'displayName': profile?.getName(),
+      'email': profile?.getEmail(),
+      'id': profile?.getId(),
+      'photoUrl': profile?.getImageUrl(),
+      'idToken': currentUser?.getAuthResponse()?.id_token,
+    };
+  }
+
   // Load the auth2 library
-  Future<void> _init(Map<String, dynamic> arguments) {
-    // 
+  GoogleAuth _init(dynamic arguments) => gapi.auth2.init(Auth2ClientConfig(
+        hosted_domain: arguments['hostedDomain'],
+        scope: arguments['scopes'].join(
+            ' '), // The backend wants a space-separated list of values, not an array
+        client_id: arguments['clientId'] ?? _autoDetectedClientId,
+      ));
+
+  Future<dynamic> _signIn(Auth2SignInOptions signInOptions) async {
+    return html.promiseToFuture<dynamic>(gapi.auth2.getAuthInstance().signIn(signInOptions));
+  }
+
+  Future<void> _signOut() async {
+    return html.promiseToFuture<void>(gapi.auth2.getAuthInstance().signOut());
   }
 
-  Future<void> _initGapi(List<void> _) {
-    // JS-interop with the global gapi method and call gapi.load('auth2'), and wait for the 
+  Future<void> _initGapi(dynamic _) {
+    // JS-interop with the global gapi method and call gapi.load('auth2'), and wait for the
     // promise to resolve...
-    
+    final Completer<void> gapiLoadCompleter = Completer<void>();
+    gapi.load('auth2', allowInterop(() {
+      gapiLoadCompleter.complete();
+    }));
+
+    return gapiLoadCompleter
+        .future; // After this is resolved, we can use gapi.auth2!
   }
 
-  /// Injects a bunch of libraries in the <head> and returns a 
+  /// Injects a bunch of libraries in the <head> and returns a
   /// Future that resolves when all load.
-  Future<List<void>> _injectJSLibraries(List<String> libraries, { Duration timeout }) {
-    final List<Future<void>> loading = <Future<void>>[
-      Future<bool>.delayed(const Duration(seconds: 10), () => true) // TODO: Remove this delay before submitting :)
-    ];
+  List<Future<void>> _injectJSLibraries(List<String> libraries,
```

Comment:
I think it makes sense for this to return a `Future<void>` by doing a `Future.wait` on `loading`

Reply:
Indeed it does. Great suggestion, thanks!